server/app/controllers/admin/AdminQuestionController.java

```diff
@@ -86,6 +95,39 @@ public CompletionStage<Result> index(Request request, Optional<String> filter) {
             classLoaderExecutionContext.current());
   }
 
+  public Result addMapFilter(Request request) {
+    Map<String, String> formData = formFactory.form().bindFromRequest(request).rawData();
+    String possibleKeysString = formData.get("possibleKeys");
+
+    // Count existing filters to determine the index for the new filter
+    long currentFilterIndex =
+        formData.keySet().stream()
+            .filter(key -> key.startsWith("filters[") && key.endsWith("].key"))
+            .count();
+
+    List<String> possibleKeysList =
+        Optional.ofNullable(possibleKeysString)
+            .map(string -> string.replaceAll("[\\[\\]]", "").trim())
+            .filter(s -> !s.isEmpty())
+            .map(
+                cleanString ->
+                    Arrays.stream(cleanString.split(","))
+                        .map(String::trim)
+                        .filter(s -> !s.isEmpty())
+                        .toList())
+            .orElse(List.of());
+
+    return ok(mapQuestionSettingsFiltersPartialView.render(
+            request,
+            new MapQuestionSettingsFiltersPartialViewModel(possibleKeysList, currentFilterIndex)))
+        .as(Http.MimeTypes.HTML);
+  }
+
+  public Result deleteMapFilter(Request request) {
+    // Return empty response - HTMX will remove the element from DOM
+    return ok("").as(Http.MimeTypes.HTML);
+  }
```

Comment:
I don't quite understand what's going on here. Are you just adding an element to the screen? That might be easier with a little javascript.

Also this is very fragile relying on the naming here to determine how to count the keys. You should create a class and have the formfactory bind to it `formFactory.form(NewClassName.class).bindFromRequest(request)`. It has to be a class, Play can't bind records.